app/operations/tests/start_operation.rb

```diff
@@ -0,0 +1,16 @@
+module Tests
+  class StartOperation
+    def initialize(test, current_user)
+      @test = test
+      @current_user = current_user
+    end
+
+    def call
+      @current_user.tests << @test unless @test.in? @current_user.tests
+      @current_user.technologies << @test.technology unless @test.technology.in? @current_user.technologies
+      @test_result = @current_user.test_results.find_by(test_id: @test.id)
+      @test_result.result = 0
```

Comment:
What is 0? Is it some sort of flag? If so what's 1?

Reply:
> What is 0? Is it some sort of flag? If so what's 1?

I decided to count right answers. In all tests we have 10 questions so it's easy to check the results. And when user'll start a new test, 'result' would be reset.